### PR TITLE
[README] Add a section on templates

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
We have some new engineers that are doing some templating. I would like to add some of the information on the templating process to our README. 

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,13 +77,147 @@ on VA.gov, you'll need to build these static pages using the following commands:
...
+
+There is also an HTML comment at the top of the file to help describe the current page.
+
+![HTML comment describing the page](.github/images/cms-html-comment.png)
+
+</details>
+
+#### Resources
+
+The implementation of the Liquid templating language used by our project is [TinyLiquid](https://github.com/leizongmin/tinyliquid). I highly recommend using [TinyLiquid's README](https://github.com/leizongmin/tinyliquid/blob/cebcb26c3839e725cf0469dccc0073799902a020/README_en.md) as your resource while writing a template, as there are small [differences from Shopify's language spec](https://github.com/leizongmin/tinyliquid#the-difference-with-liquid-language).
```
Comment (ncksllvn) [inline README.md:219]: I opened this PR solely to get a link to the TinyLiquid filter spec into our README.

File: README.md [hunk 2]
```diff
@@ -77,13 +77,147 @@ on VA.gov, you'll need to build these static pages using the following commands:
...
+  <li>Disney Cars</li>
+  <li>Toy Story</li>
+</ul>
+```
+
+</details>
+
+##### How templating works with CMS data
+
+After set up, your project should have issued a network request to fetch data from the VA CMS, Drupal. To verify, the response JSON should be visible in your local `vets-website/.cache/localhost/drupal` directory. To refresh the content, run `yarn build:content --pull-drupal`. The `vets-website/.cache` directory will be emptied and a new network request will be issued.
```
Comment (kelsonic) [inline README.md:136]: We may want to add a note that you have to have your 🧦 on when running `--pull-drupal`

File: README.md [hunk 3]
```diff
@@ -77,13 +77,147 @@ on VA.gov, you'll need to build these static pages using the following commands:
...
+
+The JSON file returned by the CMS will be parsed into separate pages during the `vets-website` compilation process. Consider opening the `vets-website/.cache/localhost/drupal/pages.json` file to explore the data structures.
```
Comment (kelsonic) [inline README.md:138]: Love this! 🔥

File: README.md (hunk 1 above)
Comment (kelsonic) [reply README.md:219]: hahaha backlinks for dayz

**Review by kelsonic — APPROVED** [review]
Thanks so much for taking the time to improve our docs! 🔥 Looks great!

**Comment by kelsonic** [comment]
This PR breaks 8 tests 🤣

**Review by U-DON — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -77,13 +77,147 @@ on VA.gov, you'll need to build these static pages using the following commands:
...
+
+For example, pretend this is a `.md` in `vagov-content.`
+
+```markdown
+---
+layout: favorite-movies.html
+movies:
+  - Ace Ventura
+  - Disney Cars
+  - Toy Story
```
[the remark this answers is not in the document]
Comment (ncksllvn) [reply README.md:104]: Haha! I will take any movie suggestions. For some reason these were the first movies I thought of

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ncksllvn) [reply README.md:221]: Good eyes!

**Pull request merged**